app/design/frontend/base/default/template/somethingdigital/criteo/basket.phtml

```diff
@@ -2,15 +2,32 @@
 
 <?php
     echo $this->getChildHtml('somethingdigital_criteo_device_detection');
+    $email = Mage::helper('somethingdigital_criteo')->getHashedEmail();
+    $items = $this->getCriteoViewBasket();
+    $useDataLayer = Mage::getStoreConfig('somethingdigital_criteo/settings/use_datalayer');
 ?>
-
-<script type="text/javascript">
-    window.criteo_q = window.criteo_q || [];
-    window.criteo_q.push(
-        { event: "setAccount", account: "<?php echo Mage::getStoreConfig('somethingdigital_criteo/settings/account_number'); ?>" },
-        { event: "setHashedEmail", email: ["<?php echo Mage::helper('somethingdigital_criteo')->getHashedEmail();?>"] },
-        { event: "setSiteType", type: deviceType },
-        { event: "viewBasket", item: <?php echo $this->getCriteoViewBasket(); ?> }
-    );
-</script>
-<?php endif; ?>
\ No newline at end of file
+    <?php if (!$useDataLayer): ?>
+        <script type="text/javascript">
+            window.criteo_q = window.criteo_q || [];
+            window.criteo_q.push(
+                { event: "setAccount", account: "<?php echo Mage::getStoreConfig('somethingdigital_criteo/settings/account_number'); ?>" },
+                { event: "setHashedEmail", email: ["<?php echo $email;?>"] },
+                { event: "setSiteType", type: deviceType },
+                { event: "viewBasket", item: <?php echo $items ?> }
+            );
+        </script>
+    <?php else: ?>
+        <script type="text/javascript">
+            var dataLayer = dataLayer || [];
+            dataLayer.push(
+                {
+                    event: "crto_basketpage",
+                    crto: {
+                        email: "<?php echo $email; ?>",
```

Comment:
Would call this `hashedEmail`

Reply:
Called email http://guides.criteotilt.com/tag-manager/gtm/?accountid=52244&advertiserid=47350&advertisername=Baked+by+Melissa&utm_medium=52244&homepageurl=http%3A%2F%2Fwww.bakedbymelissa.com%2F&accuratekey=bakedbymelissaus%3B7815412%3B7815413%3B52244#homepage